CardiacDeviceSimulator/CardiacDeviceSimulatorUtils/devices.py

```diff
@@ -0,0 +1,199 @@
+    self.harmonyTcpvDistalCurvedLengthSliderWidget = self.addSlider("Distal curved length", "Pulmonary artery side",
+                                                                    8.8, "mm", 0, 17.7 * scale, 0.1, 1)
+    self.harmonyTcpvMidRadiusSliderWidget = self.addSlider("Middle radius", "", 11, "mm", 3, 11 * scale, 0.1, 1)
+    self.harmonyTcpvMidLengthSliderWidget = self.addSlider("Middle length", "", 17.7, "mm", 5, 17.7 * scale, 0.1, 1)
+    self.harmonyTcpvProximalCurvedRadiusSliderWidget = self.addSlider("Proximal curved radius", "Pulmonary artery side",
+                                                                      21, "mm", 0, 21.5 * scale, 0.1, 1)
+    self.harmonyTcpvProximalCurvedLengthSliderWidget = self.addSlider("Proximal curved length", "Pulmonary artery side",
+                                                                      8.8, "mm", 0, 17.7 * scale, 0.1, 1)
+    self.harmonyTcpvProximalStraightRadiusSliderWidget = self.addSlider("Proximal straight radius",
+                                                                        "Right ventricle side", 21.5, "mm", 10,
+                                                                        21.5 * scale, 0.1, 1)
+    self.harmonyTcpvProximalStraightLengthSliderWidget = self.addSlider("Proximal straight length",
+                                                                        "Right ventricle side", 8.9, "mm", 0,
+                                                                        17.7 * scale, 0.1, 1)
+
+    self.logic.modelInfo[self.NAME] = {'getProfilePointsMethod': self.getProfilePoints,
+                                            'segments': ['distal', 'middle', 'proximal'],
+                                            'interpolationSmoothness': -1.0, 'parameters': {}}
+    self.ucb()
+
+  def ucb(self):
+    self.logic.setModelParameters(self.NAME, {
+      'distalStraightRadiusMm': self.harmonyTcpvDistalStraightRadiusSliderWidget.value,
+      'distalStraightLengthMm': self.harmonyTcpvDistalStraightLengthSliderWidget.value,
+      'distalCurvedRadiusMm': self.harmonyTcpvDistalCurvedRadiusSliderWidget.value,
+      'distalCurvedLengthMm': self.harmonyTcpvDistalCurvedLengthSliderWidget.value,
+      'midRadiusMm': self.harmonyTcpvMidRadiusSliderWidget.value,
+      'midLengthMm': self.harmonyTcpvMidLengthSliderWidget.value,
+      'proximalCurvedRadiusMm': self.harmonyTcpvProximalCurvedRadiusSliderWidget.value,
+      'proximalCurvedLengthMm': self.harmonyTcpvProximalCurvedLengthSliderWidget.value,
+      'proximalStraightRadiusMm': self.harmonyTcpvProximalStraightRadiusSliderWidget.value,
+      'proximalStraightLengthMm': self.harmonyTcpvProximalStraightLengthSliderWidget.value
+    })
+
+    if self._updateCB:
+      self._updateCB()
+
+  def getProfilePoints(self, params, segment=None, openSegment=True):
+    import math
+
+    #TODO: the following variable is not used at all
```

Comment:
@lassoan Is there any reason, why lengthMm was assigned but never used?

Reply:
Probably was used at some point but not anymore. You can remove it.